src/main/java/org/gradle/profiler/hp/HonestProfilerControl.java

```diff
@@ -35,20 +35,34 @@
 
     private final HonestProfilerArgs args;
     private final ScenarioSettings scenarioSettings;
+    private boolean recordedBefore = false;
 
     public HonestProfilerControl(final HonestProfilerArgs args, ScenarioSettings scenarioSettings) {
         this.args = args;
         this.scenarioSettings = scenarioSettings;
     }
 
     @Override
-    public void start() throws IOException, InterruptedException {
+    public void startSession() throws IOException, InterruptedException {
+
+    }
+
+    @Override
+    public void startRecording() throws IOException, InterruptedException {
+        if (recordedBefore) {
+            throw new RuntimeException("Recording multiple iterations with cleanup runs in between is not supported by Honest Profiler");
```

Comment:
it would be good if this were somehow applied as a decoration over those implementations that can only start and stop once.

Reply:
Extracted into a common superclass, so they only need to implement two methods instead of four.

Reply:
It would be nice to provide an option to _not_ fail if the profiler doesn't support pausing: we can still filter out some frames after recording, to remove the cleaning steps, for example.

Reply:
You could change the implementation to actually support pausing instead of failing. I.e. you could remember the pause/resume times and filter frames that fall into the paused time. I just did the minimum work to port the existing implementation.